tools/scripts/download-dep.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+import argparse
+import logging
+import os
+import shutil
+import subprocess
+import sys
+import urllib.parse
+import urllib.request
+import uuid
+from pathlib import Path
+
+# Setup logging
+# Create logger
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+# Setup console logging
+logging_console_handler = logging.StreamHandler()
+logging_formatter = logging.Formatter("%(asctime)s [%(levelname)s] %(message)s")
+logging_console_handler.setFormatter(logging_formatter)
+logger.addHandler(logging_console_handler)
+
+
+def main(argv):
+    args_parser = argparse.ArgumentParser(description="Download dependency.")
+    args_parser.add_argument("source_url", help="Url to the source file.")
+    args_parser.add_argument("source_name", help="Name of the source file.")
+    args_parser.add_argument("dest_dir", help="Destination to output the downloaded files.")
+    args_parser.add_argument("--extract", action="store_true", help="Extract the source file.")
+    args_parser.add_argument(
+        "--no-submodule",
+        action="store_false",
+        dest="use_submodule",
+        help="Do not use git submodule update",
+    )
```

Comment:
Isn't it sufficient to check if `.git` exists in the directory? I.e., do we really need `--no-submodule`?

Reply:
antlr and sqlite3 are two special cases
https://github.com/y-scope/clp/pull/508/files#diff-925c4a1e0b1dcebbc8b6a2f567f340ffd8737a7ef492ab836fc89def730ef83fL15